### DISCUSSION TITLE
How to add caption to existing image

### DISCUSSION THREAD

**Discussion by crcucb** [ImageMagick/ImageMagick#4067] [Help]
"C:\Program Files\ImageMagick-7.1.0-Q16-HDRI\convert.exe" -font Times-New-Roman-Bold-Italic -pointsize 32 -size 320x -fill navy -gravity Center caption:'test'  "C:\ProgramData\Background Image\testin.jpg"  "C:\ProgramData\Background Image\testout.jpg" 

My intent is to add caption to an existing image.  The above command is creating two output files. One named testout-1.jpg that is just a copy of testin.jpg, and the other is testout-0.jpg which is just a jpg file that has the word test

**Comment by crcucb** [comment]
I found it's adding the -composite parameter.

**Comment by fmw42** [comment]
You need to set the background color for caption:  -fill will set the text color. For ImageMagick 7, use magick.exe, not convert.exe. There is no composite parameter in your command.

Specify -size then -background right after magick.exe. The after those put your other arguments. 

Caption does not take an input image. If you want to composite with an input image, then use parentheses process to build your caption and specify your input outside the parentheses.
